Assets/Scripts/Fire.cs

```diff
@@ -38,6 +39,13 @@ public class Fire : MonoBehaviour {
                 sh.scale -= Vector3.ClampMagnitude(Vector3.one, speedDecreasing * Time.deltaTime);
                 em.rateOverTime = (ParticleSystem.MinMaxCurve)(System.Math.Pow(sh.scale.magnitude, 3) * densityFire);
             }
+
+
+            var playerCollids = Physics.OverlapSphere(transform.position, sh.scale.magnitude);
```

Comment:
You know i am not very keen on having a 3D collision calculation every frame, but maybe in this case it can be avoided - first check the distance to the player and only if it is small enough do the OverlapSphere call